### DISCUSSION TITLE
pop-up notification, when rndis device with webserver is connected

### DISCUSSION THREAD

**Discussion by Mowglli** [hathach/tinyusb#2048] [General]
Hello, i have been looking at a example, where the webusb makes a pop-up notification on the pc when connected. The example is the following project: 
[USB-dongle](https://github.com/espressif/esp-iot-solution/tree/usb/add_usb_solutions/examples/usb/device/usb_dongle)

I would like similar behavior, with a rndis device with a webserver. 
I have the webserver up and running. Put how do i make edge or similar make a pop-up? 
best regards.

**Comment by manuelbl** [comment]
The pop-up is a deliberate feature of the WebUSB specification. And WebUSB devices are specifically designed to be used by the browser and the browser only.

RNDIS is a completely different. It adds network connectivity to the computer, which is available to the operating system and all applications.

The best thing I can think of is a composite device implementing both RNDIS and WebUSB.

**Reply by Mowglli** [reply]
Thank you sir
